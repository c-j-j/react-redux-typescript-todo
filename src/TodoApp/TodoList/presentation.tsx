import * as React from 'react';

export interface Props {
  items: string[];
}

export function Presentation({items}: Props) {
  return (
    <section data-id="todo-items">
      {items.map((item) => (<p>{item}</p>))}
    </section>
  );
}