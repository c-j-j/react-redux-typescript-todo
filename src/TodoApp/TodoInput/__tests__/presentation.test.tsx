import * as React from 'react';
import { shallow } from 'enzyme';
import { Presentation } from '../presentation';
import * as renderer from 'react-test-renderer';

describe('TodoInput', () => {
  const addTodoSpy = jest.fn();

  it('adds an item', () => {
    const wrapper = shallow(<Presentation addTodo={addTodoSpy}/>);
    wrapper.find('input[type="text"]').simulate('change', {target: {value: 'foo'}});
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(addTodoSpy.mock.calls.length).toBe(1);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<Presentation addTodo={addTodoSpy}/>).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});