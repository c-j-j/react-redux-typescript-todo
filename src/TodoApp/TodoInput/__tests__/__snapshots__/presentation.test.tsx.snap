// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Enter todo item"
    type="text"
  />
</form>
`;
